spec/view_component/form/label_component_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+RSpec.describe ViewComponent::Form::LabelComponent, type: :component do
+  let(:object)  { OpenStruct.new }
+  let(:form)    { form_with(object) }
+  let(:options) { {} }
+  let(:block)   { nil }
+
+  let(:component) { render_inline(described_class.new(form, object_name, :first_name, options), &block) }
+  let(:component_html_attributes) { component.css("label").first.attributes }
+
+  context "with simple args" do
+    it do
+      expect(component.to_html).to eq(
+        %(<label for="user_first_name">First name</label>)
+      )
+    end
+  end
+
+  context "with content and options" do
+    let(:options) { { class: "custom-label" } }
+    let(:component) { render_inline(described_class.new(form, object_name, :first_name, "Your first name", options)) }
+
+    it do
+      expect(component.to_html).to eq(
+        %(<label class="custom-label" for="user_first_name">Your first name</label>)
+      )
+    end
+  end
+
+  context "with a block" do
+    let(:block) do
+      proc do
+        "Your <strong>first name</strong>".html_safe
+      end
+    end
+
+    it do
+      expect(component.to_html).to eq(
+        %(<label for="user_first_name">Your <strong>first name</strong></label>)
+      )
+    end
+  end
+
+  # context "with a block and translation param" do
```

Comment:
These tests are commented because they don't pass yet. I will open another PR to uncomment them.